### ISSUE TITLE
Rule Change: support `Array.fromAsync` in `no-invalid-this`

### ISSUE THREAD

**Issue by fasttime** [eslint/eslint#20437]
### What rule do you want to change?

no-invalid-this

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
/* eslint no-invalid-this: "error" */

const contents = await Array.fromAsync(
    ["file1.txt", "file2.txt"],
    async function (file) {
        return await this.read(file);
    },
    { read: file => readFile(file, 'utf8') }
);
```

[**Playground link**](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWludmFsaWQtdGhpczogXCJlcnJvclwiICovXG5cbmNvbnN0IGNvbnRlbnRzID0gYXdhaXQgQXJyYXkuZnJvbUFzeW5jKFxuICAgIFtcImZpbGUxLnR4dFwiLCBcImZpbGUyLnR4dFwiXSxcbiAgICBhc3luYyBmdW5jdGlvbiAoZmlsZSkge1xuICAgICAgICByZXR1cm4gYXdhaXQgdGhpcy5yZWFkKGZpbGUpO1xuICAgIH0sXG4gICAgeyByZWFkOiBmaWxlID0+IHJlYWRGaWxlKGZpbGUsICd1dGY4JykgfVxuKTtcbiIsIm9wdGlvbnMiOnsicnVsZXMiOnt9LCJsYW5ndWFnZU9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==
)

### What does the rule currently do for this code?

A problem is reported:

```text
  6:22  error  Unexpected 'this'  no-invalid-this
```

### What will the rule do after it's changed?

No problem will be reported.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

[`Array.fromAsync`](https://github.com/tc39/proposal-array-from-async#readme) has reached TC39 stage 4 in ES2026. It follows the same signature and similar semantics as `Array.from`, so I think the rule should treat it in the same way when handling `this` in the callback function. See also #20438.

**Comment by mdjermanovic** [comment]
Makes sense to me, marking as accepted.

**Comment by fasttime** [comment]
I'm working on this.
